core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -1032,11 +1033,27 @@ private void completeFileAndJournal(LockedInodePath inodePath, CompleteFileOptio
     // determined by its size in Alluxio.
     long length = fileInode.isPersisted() ? options.getUfsLength() : inAlluxioLength;
 
+    long ufsLastModifiedMs = Constants.INVALID_TIMESTAMP_MS;
+    if (fileInode.isPersisted()) {
+      // Retrieve the UFS last modified time for this file.
+      MountTable.Resolution resolution = mMountTable.resolve(inodePath.getUri());
+      AlluxioURI resolvedUri = resolution.getUri();
+      UnderFileSystem ufs = resolution.getUfs();
+      try {
+        UfsFileStatus fileStatus = ufs.getFileStatus(resolvedUri.toString());
+        ufsLastModifiedMs = fileStatus.getLastModifiedTime();
+      } catch (IOException e) {
+        // Ignore error
+        LOG.warn("Failed to retrieve UFS file status for: " + resolvedUri, e);
```

Comment:
usually we use `e.toString()` for `WARN` messages and only print the full exception stack trace for `ERROR` logging